src/coreclr/runtime/portable/AllocFast.cpp

```diff
@@ -146,20 +146,106 @@ EXTERN_C FCDECL2(Object*, RhpNewPtrArrayFast, MethodTable* pMT, INT_PTR size)
 
 EXTERN_C FCDECL1(Object*, RhpNewFast, MethodTable* pMT)
 {
-    PORTABILITY_ASSERT("RhpNewFast is not yet implemented");
-    return nullptr;
+    FCALL_CONTRACT;
+    _ASSERTE(pMT != NULL);
+
+    Thread* thread = GetThread();
+    ee_alloc_context* cxt = thread->GetEEAllocContext();
+
+    size_t sizeInBytes = (size_t)pMT->GetBaseSize();
+    sizeInBytes = ALIGN_UP(sizeInBytes, sizeof(void*));
+
+    uint8_t* alloc_ptr = cxt->getAllocPtr();
+    _ASSERTE(alloc_ptr <= cxt->getAllocLimit());
+    if ((size_t)(cxt->getAllocLimit() - alloc_ptr) >= sizeInBytes)
+    {
+        cxt->setAllocPtr(alloc_ptr + sizeInBytes);
+        PtrArray* pObject = (PtrArray *)alloc_ptr;
+        pObject->SetMethodTable(pMT);
+        return pObject;
+    }
+
+    return AllocateObject(pMT, 0, 0);
 }
 
 EXTERN_C FCDECL1(Object*, RhpNewFastAlign8, MethodTable* pMT)
 {
-    PORTABILITY_ASSERT("RhpNewFastAlign8 is not yet implemented");
-    return nullptr;
+    FCALL_CONTRACT;
+    _ASSERTE(pMT != NULL);
+
+    Thread* thread = GetThread();
+    ee_alloc_context* cxt = thread->GetEEAllocContext();
+
+    size_t sizeInBytes = (size_t)pMT->GetBaseSize();
+    sizeInBytes = ALIGN_UP(sizeInBytes, sizeof(void*));
+
+    uint8_t* alloc_ptr = cxt->getAllocPtr();
+    bool requiresAlignObject = !IS_ALIGNED(alloc_ptr, sizeof(int64_t));
+    size_t paddedSize = sizeInBytes;
+    if (requiresAlignObject)
+    {
+        // We are assuming that allocation of minimal object flips the alignment
+        paddedSize += MIN_OBJECT_SIZE;
```

Comment:
```suggestion
        // We are assuming that allocation of minimal object flips the alignment
        paddedSize += MIN_OBJECT_SIZE;
        _ASSERTE(IS_ALIGNED(alloc_ptr + paddedSize, sizeof(int64_t)));
```

Sounds like something to assert after we made the adjustment.

Reply:
The issue here was that we would be asserting alignment for address after the new object we are align8-allocating and not the actual address of the allocated object itself.